File: base/src/main/java/com/appoets/base/di/component/BaseComponent.java
package com.appoets.base.di.component;

import com.appoets.base.base.BaseRespositary;
import com.appoets.base.di.modules.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {NetworkModule.class})
public interface BaseComponent {
    void inject(BaseRespositary respositary);
}

File: base/src/main/java/com/xjek/base/di/BaseComponent.kt
package com.xjek.base.di

import com.xjek.base.repository.BaseRepository
import com.xjek.base.utils.RunTimePermission
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [WebServiceModule::class,AppContainerModule::class])
interface BaseComponent {
    fun inject(baseRepository: BaseRepository)

    fun inject(runtimePermission:RunTimePermission)
}